### ISSUE TITLE
Switch to Singles whenever posible?

### ISSUE THREAD

**Issue by tasegula** [tbruyelle/RxPermissions#124]
Given that all the methods that return Observable<Boolean>, actually return a single response for all the requested permissions, wouldn't it be better for those methods to return Single<Boolean>, and as such showing that it will only give you one answer?

**Comment by tbruyelle** [comment]
Yes it makes sense, I thought about that too but it would introduce a breaking change without real benefit (`Single` class didn't exist when I started the library).

**Comment by PaulWoitaschek** [comment]
The benefit is that it's clear what will be returned. As .request returns an Observable I assumed that there is an update whenever that permission is granted.

I'd make a PR for that if this is wanted.

**Comment by AleksanderMielczarek** [comment]
I like this idea. What do you think of new methods like:
```
fun requestSingle() : Single<Permission>
fun requestEachCombinedSingle() : Single<Permission>
```
It'll add support for `Single` but wouldn't break backward compatibility.

**Comment by tbruyelle** [comment]
I don't want to introduce more methods to the API. It adds complexity.

**Comment by neworld** [comment]
Maybe it is a good time for major change 1.0.0 and `Single`?

**Comment by Oknesif** [comment]
I suggest replacing the return type of RxPermission.request to Single<Boolean>. @tasegula if you give me approval for this change, I am ready to make PR.
